Automattic/jetpack
# Enforce the usage of connection initial state

This PR is a proposal to remove the possibility to make an async request to fetch the connection status and rely only on the initial state provided by the connection composer package.

Background:

The async request to the connection status was created in the early stages of development when there was no store in the connection package and no unique initial state provided by the connection package.

It solves the problem but it also adds some complexities, such as having to handle the UI while the store is fetching the connection status - waiting for the request response.

Once there is now a provided initial state and we are using it everywhere, there's no benefit in keeping this option.

Reasons for the change:

* There is already an implicit dependency between the JS connection package and the Composer connection package. the JS package depends on the composer package. For example, all the REST endpoints it reaches live on the composer package;
* It is preferable to have only one clear way of doing things. The way it is no is like "you can inform the initial state to save one ajax request, but you may also not inform..."
* Even though we have a `resolver` for the `getConnectionStatus` selector, we are accessing this data directly via `select` in the selector, which means that, in the case where the initial state is not there, we may have wrong information being passed to the component and then a re-render when the async resolver returns. We are not properly using `resolveSelect` as we should

#### Changes proposed in this Pull Request:

* Enforces the usage of the Initial State provided by the composer Connection package in the JS connection package

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Jetpack plugin, Backup plugin and connection-ui should keep working as they were.
* Go to plugins/backup/src/php/class-jetpack-backup.php and comment out line 107 that adds the initial state. Reload the page and see the error in the console
